compiler_gym/datasets/datasets.py

```diff
@@ -0,0 +1,480 @@
+        from this datasets collection.
+
+        :param dataset: A :meth:`Dataset <compiler_gym.datasets.Dataset>`
+            instance, or the name of a dataset.
+
+        :return: :code:`True` if the dataset was uninstalled, :code:`False` if
+            it was already uninstalled.
+        """
+        dataset: Dataset = self.dataset(dataset)
+        uninstalled = dataset.installed
+        dataset.uninstall()
+        return uninstalled
+
+    def require(self, dataset: Union[str, Dataset]) -> bool:
+        """Require that a dataset is available for immediate use.
+
+        This performs any one-time setup of the dataset so that it can be used.
+        It is equivalent to:
+
+            >>> env.datasets.add(dataset)
+            >>> env.datasets.activate(dataset)
+            >>> env.datasets.install(dataset)
+
+        :param dataset: A :meth:`Dataset <compiler_gym.datasets.Dataset>`
+            instance, or the name of a dataset.
+
+        :return: :code:`True` if the dataset was made available, :code:`False`
+            if it was already available.
+        """
+        added = False
+        # Use an attribute check rather than isinstance() to allow mocking the
+        # Dataset class in testing.
+        if hasattr(dataset, "name"):
+            added = self.add(dataset)
+            dataset_name = dataset.name
+        else:
+            dataset_name = resolve_uri_protocol(dataset)
+        activated = self.activate(dataset_name)
+        installed = self.install(dataset_name)
+        return added or activated or installed
+
+    def benchmarks(self) -> Iterable[Benchmark]:
+        """Enumerate the (possibly infinite) benchmarks lazily.
+
+        Benchmarks order is consistent across runs. Benchmarks from each dataset
+        are returned in order. The order of :meth:`benchmarks()
+        <compiler_gym.datasets.Datasets.benchmarks>` and :meth:`benchmark_uris()
+        <compiler_gym.datasets.Datasets.benchmark_uris>` is the same.
+
+        :return: An iterable sequence of :class:`Benchmark
+            <compiler_gym.datasets.Benchmark>` instances.
+        """
+        for dataset in self.datasets():
+            yield from dataset.benchmarks()
+
+    def benchmark_uris(self) -> Iterable[str]:
+        """Enumerate the (possibly infinite) benchmark URIs.
+
+        Benchmark URI order is consistent across runs. URIs from each dataset
+        are returned in order. The order of :meth:`benchmarks()
+        <compiler_gym.datasets.Datasets.benchmarks>` and :meth:`benchmark_uris()
+        <compiler_gym.datasets.Datasets.benchmark_uris>` is the same.
+
+        :return: An iterable sequence of benchmark URI strings.
+        """
+        for dataset in self.datasets():
+            yield from dataset.benchmark_uris()
+
+    def benchmark(self, uri: Optional[str] = None) -> Benchmark:
+        """Select a benchmark.
+
+        If a benchmark URI is given, the corresponding :class:`Benchamrk
```

Comment:
Benchamrk -> Benchmark

Reply:
Thanks! Turns out I made that typo all over the place :) #199